File: java_version/hashSets/tripletSum.java
import java.util.HashSet;

public class Solution
{
  public boolean hasTripletSum(int[] arr, int target)
  {
      int n = arr.length;

      for(int i=0; i<n; i++)
        {
          HashSet<Integer> seen = new HashSer<>();
          int currTarget = target - arr[i];

          for(int j=i+1; j<n;j++)
            {
              if(seen.contains(currTarget-arr[j]))
              {
                return true;
              }

              seen.add(arr[j]);
            }
        }

       return false;
  }

File: tripletSum.py
class Solution:
    
    def hasTripletSum(self, arr, target):
        n = len(arr)
        
        # Iterate over all pairs
        for i in range(n):
            seen = set()
            curr_target = target - arr[i]
            
            # Check if there exists a pair (arr[j], arr[k]) such that arr[j] + arr[k] = curr_target
            for j in range(i+1, n):
                if (curr_target - arr[j]) in seen:
                    return True
                seen.add(arr[j])
        
        return False
